### PR TITLE
Add a Physics-using component to current components

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #24103

cutting up the old large PR for physics still.
we could delete the old FlowComponentNS or keep both, both are simply transition options, with this one being more general but having less-convenient syntax in a way

### CODE REVIEW

**Comment by GiudGiud** [comment]
I ll be able to fix NS but if the Griffin thing is related that's going to be difficult

**Comment by GiudGiud** [comment]
Ready for review imo

I raised https://github.com/libMesh/libmesh/issues/3882 for the destruction ordering issue that this PR avoids in a dirty-ish way

File: modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Operating conditions
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i:1]: Is there no test for this input file?

File: modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    energy_inlet_function = '300'
+
+    wall_boundaries = 'comp1:top comp1:bottom'
+    momentum_wall_types = 'noslip symmetry'
+    energy_wall_types = 'heatflux'
+    energy_wall_function = '1'
+
+    outlet_boundaries = 'comp1:right'
+    # momentum_outlet_types = 'fixed-pressure'
+    # pressure_function = 'f1'
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i:47]:
```suggestion
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Also, were you wanting to address the libMesh issue before this to avoid "dirtiness"?

File: modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i (hunk 1 above)
Comment (GiudGiud) [reply modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i:1]: lemme check

**Comment by GiudGiud** [comment]
> Also, were you wanting to address the libMesh issue before this to avoid "dirtiness"?

depends on what we decide to do.

File: modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i (hunk 1 above)
Comment (GiudGiud) [reply modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i:1]: deleting it

Comment (GiudGiud) [reply modules/thermal_hydraulics/test/tests/junctions/wcnsfv/two_components_same_vars.i:1]: looking at a griffin issue then I ll reupload

**Comment by joshuahansel** [comment]
> depends on what we decide to do.

I say if there's no urgency for this PR, have that conversation on the libmesh issue and make a decision on what's done there before proceeding with this PR.

**Comment by GiudGiud** [comment]
no urgency

I just feel better when I have like 1 or so PR to work on rather than 5 :D

File: framework/include/executioners/SolveObject.h [hunk 3]
```diff
@@ -41,7 +41,9 @@ class SolveObject : public MooseObject, public PerfGraphInterface, public Postpr
   /// Reference to FEProblem
   FEProblemBase & _problem;
   /// Displaced problem
-  std::shared_ptr<DisplacedProblem> _displaced_problem;
+  /// We would rather have a shared pointer here. But the displaced problem currently must be destructed
+  /// before the displaced mesh, and keeping a shared pointer here delays this
+  DisplacedProblem * _displaced_problem;
```
Comment (joshuahansel) [inline framework/include/executioners/SolveObject.h:46]: So I guess only this move to raw pointer is the "dirtiness".

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Waiting for libmesh update to avoid switching to raw pointer. Otherwise looks good.

File: framework/include/executioners/SolveObject.h (hunk 3 above)
Comment (GiudGiud) [reply framework/include/executioners/SolveObject.h:46]: it s kind of being endorsed in that discussion actually.
Alex says use less shared_ptr

Comment (joshuahansel) [reply framework/include/executioners/SolveObject.h:46]: Oh sorry I misunderstood, I thought "my current solution" meant the changes you were proposing to make in libmesh, which would avoid the changes made here.

**Review by joshuahansel — APPROVED** [review]
I misunderstood the plan that was discussed. The move to raw pointer is preferred.

File: framework/include/executioners/SolveObject.h [hunk 4]
```diff
@@ -41,7 +41,9 @@ class SolveObject : public MooseObject, public PerfGraphInterface, public Postpr
   /// Reference to FEProblem
   FEProblemBase & _problem;
   /// Displaced problem
-  std::shared_ptr<DisplacedProblem> _displaced_problem;
+  /// We would rather have a shared pointer here. But the displaced problem currently must be destructed
+  /// before the displaced mesh, and keeping a shared pointer here delays this
```
Comment (lindsayad) [inline framework/include/executioners/SolveObject.h:45]: I don't know that we would rather have a shared pointer here. I don't see any reason why any `SolveObject` (there could be many of them) should be considered an owner of the displaced problem

Comment (GiudGiud) [reply framework/include/executioners/SolveObject.h:45]: yeah this is outdated. I ll delete the comment

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review and thanks for the support! I feel like I'm walking the plank sometimes on components & physics

**Pull request merged**